### PR TITLE
Update TensorRT documentation

### PR DESCRIPTION
**Pull request by stevenlix**
1. Update TensorRT version info
2. Update configuration settings
3. Add provider option API instruction
4. Fix broken links

### CODE REVIEW

**Comment by jywu-msft** [comment]
Thanks. we also need to add a section on how to build using TRT 7.1 instead of TRT 7.2 as some customers want to support that combination.

File: docs/how-to/build/eps.md [hunk 1]
```diff
@@ -87,18 +87,18 @@ If you want to build with an earlier version, you must temporarily remove the 'C
...
 
-See more information on the TensorRT Execution Provider [here](../reference/execution-providers/TensorRT-ExecutionProvider.md).
+See more information on the TensorRT Execution Provider [here](../../reference/execution-providers/TensorRT-ExecutionProvider.md).
 
 ### Prerequisites
 {: .no_toc }
 
 * Install [CUDA](https://developer.nvidia.com/cuda-toolkit) and [cuDNN](https://developer.nvidia.com/cudnn)
-   * The TensorRT execution provider for ONNX Runtime is built and tested with CUDA 11.0 and cuDNN 8.0.
+   * The TensorRT execution provider for ONNX Runtime is built and tested with CUDA 11.1 and cuDNN 8.2.
```
Comment (jywu-msft) [inline docs/how-to/build/eps.md:96]: i think we need to change this to a support matrix (table)
where we indicate that we also support other versions of TRT/CUDA/CuDNN
e.g. 7.1.3.4/10.2/7.6.x is also valid combination

File: docs/reference/execution-providers/TensorRT-ExecutionProvider.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ With the TensorRT execution provider, the ONNX Runtime delivers better inferenci
 
 See [Build instructions](../../how-to/build/eps.md#tensorrt).
 
-The TensorRT execution provider for ONNX Runtime is built and tested with TensorRT 7.1.3.4.
+The TensorRT execution provider for ONNX Runtime is built and tested with TensorRT 7.2.2.
```
Comment (jywu-msft) [inline docs/reference/execution-providers/TensorRT-ExecutionProvider.md:25]: both 7.1.3.4 and 7.2.2 can work.

Comment (faxu) [reply docs/reference/execution-providers/TensorRT-ExecutionProvider.md:25]: Do we have tests with 7.1.3.4? Is it normally expected that this is backwards compatible or should it be explicitly called out?

**Review by faxu — APPROVED** [review]

File: docs/reference/execution-providers/TensorRT-ExecutionProvider.md (hunk 2 above)
Comment (stevenlix) [reply docs/reference/execution-providers/TensorRT-ExecutionProvider.md:25]: We don't have CI build for TRT7.1 after we moved to TRT7.2. synced with @jywu-msft offline and we probably need such CI build for TRT7.1

Comment (stevenlix) [reply docs/reference/execution-providers/TensorRT-ExecutionProvider.md:25]: @jywu-msft @faxu Please help to merge the PR. It seems I am not authorized to merge.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**